### PR TITLE
(Feat) text-tags.js -> support for custom metadata

### PR DESCRIPTION
**Pull request by Deokumarjnu**
https://github.com/quintype/ace-planning/issues/11

### CODE REVIEW

File: src/text-tags.js [hunk 1]
```diff
@@ -1,7 +1,7 @@
 import {get, isEmpty} from 'lodash';
 import {objectToTags} from './utils';
 
-function buildTagsFromStory(config, story, url = {}) {
+function buildTagsFromStory(config, story, url = {}, data = {}) {
```
Comment (nandap4790) [inline src/text-tags.js:4]: better to rename the param, so that it can be more explicit

File: src/text-tags.js [hunk 2]
```diff
@@ -27,16 +27,16 @@ function buildTagsFromStory(config, story, url = {}) {
...
   const authors = get(story, ['authors'], []).map(author => author.name);
-
+  const customSeo = get(data,["data","customSeo"], {});
   const storyMetaData = {
-    title: seo["meta-title"] || story.headline,
-    "page-title": seo["meta-title"] || story.headline,
-    description: seo["meta-description"] || story.summary,
-    keywords: (seo["meta-keywords"] || (story.tags || []).map(tag => tag.name)).join(","),
+    title: customSeo.title || seo["meta-title"] || story.headline,
+    "page-title": customSeo.title || seo["meta-title"] || story.headline,
```
Comment (nandap4790) [inline src/text-tags.js:33]: move this to a variable

File: src/text-tags.js [hunk 3]
```diff
@@ -49,12 +49,12 @@ function buildTagsFromStory(config, story, url = {}) {
   return storyMetaData;
 }
 
-function buildTagsFromTopic(config, tag, url = {}) {
+function buildTagsFromTopic(config, tag, url = {}, data) {
```
Comment (nandap4790) [inline src/text-tags.js:52]: same as above

File: src/text-tags.js [hunk 4]
```diff
@@ -73,12 +73,13 @@ function buildTagsFromTopic(config, tag, url = {}) {
...
-function buildTagsFromAuthor(config, author, url = {}) {
+function buildTagsFromAuthor(config, author, url = {}, data) {
   if(isEmpty(author)) return;
 
-  const authorName = author.name;
+  const customSeo = get(data,["data","customSeo"], {})
+  const authorName = customSeo.title || author.name;
   const authorUrl = `${config['sketches-host']}${url.pathname}`;
-  const description = author.bio || `View all articles written by ${authorName}`;
+  const description = customSeo.description || author.bio || `View all articles written by ${authorName}`;
```
Comment (nandap4790) [inline src/text-tags.js:82]: how are we going to differentiate in the app level for different seo tags if every key is a title/description, etc?

**Review by nandap4790 — CHANGES_REQUESTED** [review]

File: src/text-tags.js (hunk 1 above)
Comment (Deokumarjnu) [reply src/text-tags.js:4]: the data here represent all the data that is coming by the response, it does not only have SEO data but it has config, API response, etc. so I'm calling it as data.

File: src/text-tags.js (hunk 4 above)
Comment (nandap4790) [reply src/text-tags.js:91]: ??

**Review by nandap4790 — CHANGES_REQUESTED** [review]

File: src/text-tags.js (hunk 4 above)
Comment (Deokumarjnu) [reply src/text-tags.js:91]: I'm differentiating it with a normal description will have a description as a keyword while ogDescription should have a description as a key. here is [doc](https://developers.quintype.com/malibu/tutorial/custom-seo-metadata.html)

**Review by nandap4790 — APPROVED** [review]

**Pull request merged**